pinot-common/src/main/java/org/apache/pinot/common/function/FunctionRegistry.java

```diff
@@ -106,12 +156,16 @@ public static void registerFunction(Method method, boolean nullableParameters) {
   /**
    * Registers a method with the given function name.
    */
-  public static void registerFunction(String functionName, Method method, boolean nullableParameters) {
-    FunctionInfo functionInfo = new FunctionInfo(method, method.getDeclaringClass(), nullableParameters);
+  public static void registerFunction(String functionName, Method method, boolean nullableParameters, boolean narrowing,
```

Comment:
qq: how do we plan to generate a signature for calcite? 
- for example if a function accepts narrowing, i am not 100% sure how to tell calcite's catalog reader that a function accepts multiple parameter type (let a long the cartesian combination of all possible variations)
- it looks like we might need to use direct override of the planner function matching algorithm ?

Reply:
see comment above, Calcite has a functionality to support type coercion, which will basically allow you to cast nearly any sql type to another sql type if you pass in `coerce = true`